src/indexing.jl

```diff
@@ -0,0 +1,86 @@
+"""
+    CheckpointOutput
+
+This is a index entry describing the output file from a checkpoint.
+You can retrieve a list of these from a folder full of such outputs, using
+[`index_checkpoint_files`](@ref).
+
+For accessing details of the CheckpointOutput the following helpers are provided:
+[`path`](@ref), [`name`](@ref), [`groups`](@ref), [`tags`](@ref).
+Further: `getproperty` is overloaded so that you can access the value of the tag `:foo` via
+`x.foo`.
+"""
+struct CheckpointOutput{P<:AbstractPath}
+    path::P
+    name::AbstractString
+    groups::NTuple{<:Any, AbstractString}
+    tags::NTuple{<:Any, Pair{Symbol, <:AbstractString}}
```

Comment:
Oh this is nicer, do you mind changing this line to be consistent across the package?
https://github.com/invenia/Checkpoints.jl/blob/d8c15b97789962b541711a0f063840d9bb59924c/src/Checkpoints.jl#L28

Reply:
is it? those are the basically same

Reply:
yeah sure - yours reads more nicely

Reply:
I think mine might break, because i contrain them all to be the same abstract sttring subtype, which I know will work in this case, because they are all SubStrings, but might not in others.